### PR TITLE
AmbientCapabilities in unbound.service.in and futher hardenings

### PR DESCRIPTION
**Pull request by DavidOsipov**
Main change is an intro of AmbientCapabilities, which allow to bind to privileged ports (1-1024) without super-user privileges, thus it's possible to further harden the process.

I'm testing other hardening options, probably will contribute soon.

### CODE REVIEW

File: contrib/unbound.service.in [hunk 1]
```diff
@@ -74,7 +76,8 @@ StateDirectory=unbound
 RestrictAddressFamilies=AF_INET AF_INET6 AF_NETLINK AF_UNIX
 RestrictRealtime=true
 SystemCallArchitectures=native
-SystemCallFilter=~@clock @cpu-emulation @debug @keyring @module mount @obsolete @resources
+SystemCallFilter=@system-service
+SystemCallFilter=~@clock @cpu-emulation @debug @keyring @module @mount @obsolete @resources @privileged @debug @reboot
```
Comment (gthess) [inline contrib/unbound.service.in:80]: Isn't this redundant now since you only allow `@system-service` above?

File: contrib/unbound.service.in [hunk 2]
```diff
@@ -55,6 +55,8 @@ ExecStart=@UNBOUND_SBIN_DIR@/unbound -d -p
 NotifyAccess=main
 Type=notify
 CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_SETGID CAP_SETUID CAP_SYS_CHROOT CAP_SYS_RESOURCE CAP_NET_RAW
+CapabilityBoundingSet=~CAP_SYS_ADMIN
```
Comment (gthess) [inline contrib/unbound.service.in:58]: Isn't this implied because `CAP_SYS_ADMIN` is not defined above?

**Review by gthess — CHANGES_REQUESTED** [review]
Hi, thanks for this. I have a few comments though.
@Maryse47 could you have a second look since you have contributed a lot in this file?

**Comment by Maryse47** [comment]
The service still starts as root so I don't see actual hardening effect of this change. `SystemCallFilter` and `CapabilityBoundingSet` changes should be explained somewhere.

File: contrib/unbound.service.in (hunk 1 above)
Comment (DavidOsipov) [reply contrib/unbound.service.in:80]: Not really, here `@system-service` is a wide concept, which is allowed. And the line SystemCallFilter=~ restricts calls outside and inside `@system-service` concept. The bad thing is the [documentation ](https://www.freedesktop.org/software/systemd/man/systemd.exec.html) doesn't tell what is really hidden behind the `@system-service`. 

If you have any info on redundancy, please share :)

**Comment by DavidOsipov** [comment]
> The service still starts as root so I don't see actual hardening effect of this change. `SystemCallFilter` and `CapabilityBoundingSet` changes should be explained somewhere.

Ok, let me refine the changes and I'll open another pull request.

**Comment by Maryse47** [comment]
Note that starting the service as non-root isn't an option either because it would break existing setups. I think those changes belong to downstream where you control specific execution environment.

File: contrib/unbound.service.in (hunk 1 above)
Comment (Maryse47) [reply contrib/unbound.service.in:80]: So undocumented feature being introduced without explanation :smile:

I think this describes the problem with this change. Replacing `SystemCallFilter=~XXX` with `SystemCallFilter=@system-service` may make sense (this is what systemd-resolved uses) although there is a slight risk of regressing something. I don't see reason for using both at same time which makes hard to understand the net effect and stacking broad seccomp filters may introduce perf penalties.

**Comment by Maryse47** [comment]
There is alternative [portable](https://github.com/NLnetLabs/unbound/blob/master/contrib/unbound_portable.service.in) service which may be better target for running unprivileged.

**Comment by DavidOsipov** [comment]
> There is alternative [portable](https://github.com/NLnetLabs/unbound/blob/master/contrib/unbound_portable.service.in) service which may be better target for running unprivileged.

Thanks! I'll try to play around with it and yeap, running unbound as unprivileged service is a headache

**Pull request closed without merging**